peakina/readers/excel.py

```diff
@@ -0,0 +1,133 @@
+"""
+Module to add xml support
+"""
+import enum
+from io import StringIO
+from typing import Any, List, Optional
+
+import pandas as pd
+import xlrd
+from openpyxl import load_workbook
+from openpyxl.utils.exceptions import InvalidFileException
+
+from peakina.readers.csv import PreviewArgs
+
+
+class EXCEL_TYPE(enum.Enum):
+    NEW = "new"
+    OLD = "old"
+
+
+def _read_old_xls_format(wb: Any, sh_name: str, preview: Optional[PreviewArgs]) -> Any:
+    """ """
+    if preview:
+        to_iter = range(
+            preview.get("offset", 0), preview.get("offset", 0) + preview.get("nrows", 500) + 1
```

Comment:
maybe we should switch `Preview` to a `dataclass` (or a pydantic class) with default values.
So it's either `None` or a `Preview` instance. WDYT?

Reply:
Good idea, this is how i did : https://github.com/ToucanToco/peakina/pull/69/commits/0a943458354f50442209faed94ab5e85bd6f9fb3
Let me know what you think !